analyzers/src/SonarAnalyzer.CSharp/Extensions/SyntaxNodeExtensions.cs

```diff
@@ -91,6 +91,7 @@ public static bool IsPartOfBinaryNegationOrCondition(this SyntaxNode node)
                 SyntaxKind.StructDeclaration => "struct",
                 SyntaxKind.InterfaceDeclaration => "interface",
                 SyntaxKindEx.RecordClassDeclaration => "record",
+                SyntaxKindEx.RecordStructDeclaration => "record struct",
```

Comment:
Didn't we see something like this already in `ClassAndMethodName.cs`?

Reply:
#5571 with `ClassAndMethodName` used a dictionary field locally to do the same thing. The PR is already merged. I took a note about that and will create a follow-up PR later so ClassAndMethodName uses the `GetDeclarationTypeName` extension method instead.